### PR TITLE
Rename clear* methods to remove* methods.

### PR DESCRIPTION
**Pull request by hsorby**
Addresses #448

### CODE REVIEW

**Comment by hsorby** [comment]
This one is now ready for review.

File: src/bindings/interface/logger.i [hunk 1]
```diff
@@ -10,7 +10,7 @@
 %feature("docstring") libcellml::Logger::addError
 "Adds an error to this logger.";
 
-%feature("docstring") libcellml::Logger::clearErrors
+%feature("docstring") libcellml::Logger::removeAllErrors
```
Comment (kerimoyle) [inline src/bindings/interface/logger.i:13]: What's the rationale behind `removeAllErrors` - are we able to remove them selectively eg with a `removeError(index)` function?  That could be useful if/when we get to having different levels or types of error.  If not, I would prefer it just be `removeErrors`

**Review by kerimoyle — COMMENTED** [review]
Can we please have a removeMath method for components too (as in #443) or is that coming later?

File: src/bindings/interface/logger.i (hunk 1 above)
Comment (hsorby) [reply src/bindings/interface/logger.i:13]: The rationale is that when we are removing all types of thing from an object we have used `removeAll*` to do that job.

- `removeAllUnits`
- `removeAllComponents`
- `removeAllResets`
- `removeAllEquivalences`
- `removeAllVariables`

Comment (hsorby) [reply src/bindings/interface/logger.i:13]: There is currently no functionality to remove/take selective errors.

**Comment by hsorby** [comment]
> Can we please have a removeMath method for components too (as in #443) or is that coming later?

It is coming in #450 which is built upon this pull request so will come in later.  This pull request is only dealing with the renaming of methods.

File: src/bindings/interface/logger.i (hunk 1 above)
Comment (kerimoyle) [reply src/bindings/interface/logger.i:13]: Ok, thanks.

**Review by kerimoyle — APPROVED** [review]
All good with me.

**Review by agarny — APPROVED** [review]

**Pull request merged**